src/services/event.ts

```diff
@@ -444,3 +445,112 @@ export async function findAllEvents(): Promise<EventShape[]> {
     raw: true,
   });
 }
+
+const splitPipe = (str: string) => str.split('\n').map((s) => s.trim()).filter(Boolean);
+
+const mappings: Record<string, string> = {
+  Audience: 'audience',
+  Creator: 'creator',
+  'Edit Title': 'eventName',
+  'Event Duration/#NC Days of Support': 'eventDuration',
+  'Event ID': 'eventId',
+  'Overall Vision/Goal for the PD Event': 'vision',
+  'Reason for Activity': 'reasons',
+  'Target Population(s)': 'targetPopulations',
+  'Event Organizer - Type of Event': 'eventOrganizer',
+};
+
+const toSplit = ['targetPopulations', 'reasons'];
+
+const replacements: Record<string, string> = {
+  'Preschool (ages 3-5)': 'Preschool Children (ages 3-5)',
+  'Pregnant Women/Pregnant People': 'Pregnant Women / Pregnant Persons',
+  'Pregnant Women': 'Pregnant Women / Pregnant Persons',
+};
+
+const applyReplacements = (value: string) => replacements[value] || value;
+
+const mapLineToData = (line: Record<string, string>) => {
+  const data: Record<string, unknown> = {};
+
+  Object.keys(line).forEach((key) => {
+    const mappedKey = mappings[key] || key;
+    data[mappedKey] = toSplit.includes(mappedKey)
+      ? splitPipe(line[key]).map(applyReplacements)
+      : line[key];
+  });
+
+  return data;
+};
+
+const checkUserExists = async (creator: string) => {
+  const user = await db.User.findOne({ where: { email: creator } });
```

Comment:
Do we want to verify the user can Read/Write training reports?

Reply:
This is a good question to pose to a wider audience - maybe @ standup

Reply:
We by convention always check permissions in the handlers, not in the services. If this is called in the admin namespace, then a user will be authenticated as an admin before calling this.

Reply:
deleted my comment as it was dumb

Reply:
We confirmed in standup that we need to not import if the user defined in the line item doesn't have write permissions. This is complete and tests are written for it